sale_stock_restocking_fee_invoicing/models/stock_move.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from collections import defaultdict
+
+from odoo import fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    charge_restocking_fee = fields.Boolean(
+        help="If checked your customer will be charged for accepting "
+        "returned goods.",
+        default=False,
+        readonly=True,
+    )
+    picking_kind = fields.Selection(related="picking_id.picking_kind", readonly=True)
+
+    def _action_done(self, cancel_backorder=False):
+        result = super()._action_done(cancel_backorder=cancel_backorder)
+        move_ids_to_charge_by_so = defaultdict(list)
+        for move in self:
+            if not move._is_restocking_fee_chargeable():
+                continue
+            so = move.sale_line_id.order_id
```

Comment:
This can be void.

Either you check if order_id is set here, either you add the condition in the _is_restocking_fee_chargeable().

It causes a stack trace :

![image](https://user-images.githubusercontent.com/19529533/233371878-db63d39f-0c05-461f-9620-3072100252d1.png)